raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -126,9 +126,14 @@
  *    gracefully resign from the current epoch. This causes remaining voters to immediately
  *    begin a new election.
  *
- * 4) {@link FetchRequestData}: This is the same as the usual Fetch API in Kafka, but we piggyback
- *    some additional metadata on responses (i.e. current leader and epoch). Unlike partition replication,
- *    we also piggyback truncation detection on this API rather than through a separate truncation state.
+ * 4) {@link FetchRequestData}: This is the same as the usual Fetch API in Kafka, but we add snapshot
+ *    check before responding, and we also piggyback some additional metadata on responses (i.e. current
+ *    leader and epoch). Unlike partition replication, we also piggyback truncation detection on this API
+ *    rather than through a separate truncation state.
+ *
+ * 5) {@link FetchSnapshotRequestData}: Sent by the follower when FetchResponse include a snapshot id,
+ *    this is similar to the Fetch API since the snapshot is also stored as FileRecords, but we use
+ *    {@link UnalignedRecords} in FetchSnapshotResponse because the records is not necessarily offset-aligned.
```

Comment:
Thanks for doing this. I also noticed it was missing and fixed it in this PR: https://github.com/apache/kafka/pull/10085/files#diff-1da15c51e641ea46ea5c86201ab8f21cfee9e7c575102a39c7bae0d5ffd7de39R134-R137

Maybe reconcile the two changes and we can merge this one.

Reply:
Thank you for your advice, Done.